From 3986ca8be0b34c83351bbb50783c85d034fb2a19 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Mon, 16 Nov 2020 15:51:28 +0100
Subject: [PATCH 5/9] Revert "nvme: fix incorrect behavior when BLKROSET is
 called by the user"

This reverts commit d1dd6ef1461b0d832a8b73ab69d8ae2a06a751db.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 drivers/nvme/host/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/nvme/host/core.c b/drivers/nvme/host/core.c
index 349fba056..b130696b0 100644
--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -2064,6 +2064,8 @@ static void nvme_update_disk_info(struct gendisk *disk,
 
 	if (id->nsattr & NVME_NS_ATTR_RO)
 		set_disk_ro(disk, true);
+	else
+		set_disk_ro(disk, false);
 }
 
 static inline bool nvme_first_scan(struct gendisk *disk)
-- 
2.29.2.260.ge31aba42fb

